File: app/src/types/report.ts
export interface Report {
  id: number;
  latitude: number;
  longitude: number;

  original_image_url: string;
  annotated_image_url: string;

  garbage_detected: boolean;
  garbage_count: number;

  highest_confidence: number;

  status: string;

  created_at: string;
}

File: server/models/report.py
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy import (
    Column,
    Integer,
    Float,
    Boolean,
    String,
    DateTime
)
from datetime import datetime


class Base(DeclarativeBase):
    pass


class Report(Base):

    __tablename__ = "reports"

    id = Column(Integer, primary_key=True)

    latitude = Column(Float, nullable=False)

    longitude = Column(Float, nullable=False)

    original_image_path = Column(String, nullable=False)

    annotated_image_path = Column(String, nullable=True)

    garbage_detected = Column(Boolean, nullable=False)

    garbage_count = Column(Integer, nullable=False)

    highest_confidence = Column(Float, nullable=False)

    status = Column(String, default="PENDING")

    created_at = Column(
        DateTime,
        default=datetime.utcnow
    )